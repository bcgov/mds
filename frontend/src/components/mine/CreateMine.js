import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Card, Form, Input, Button, notification } from 'antd';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

import { getUserAccessData } from '@/selectors/authenticationSelectors';
import { CreateGuard } from '@/HOC/CreateGuard';
import { createMineRecord } from '@/actionCreators/mineActionCreator';
import AddMineRecordForm from './Forms/AddMineRecordForm';
import * as routes from '@/constants/routes';

const propTypes = {
  createMineRecord: PropTypes.func.isRequired,
  userRoles: PropTypes.array.isRequired,
};

const defaultProps = {
  createMineRecord: {},
  userRoles: [],
};

export class CreateMine extends Component {
  constructor(props) {
    super(props);
    this.state = {
      redirectTo: null
    };
  }

  handleSubmit = (value) => {
    this.props.createMineRecord(value.mineName).then(() => {
      this.setState({redirectTo: routes.MINE_DASHBOARD.route})
    })
  }

  render() {
    if (this.state.redirectTo) {
      return <Redirect push to={this.state.redirectTo} />
    }
    return (
      <div>
        <h1>Create A Mine Record</h1>
        <Card title="Create Mine Form">
          <AddMineRecordForm onSubmit={this.handleSubmit} />
        </Card>
      </div>
    );
  }
}

CreateMine.propTypes = propTypes;
CreateMine.defaultProps = defaultProps;

const mapStateToProps = (state) => {
  return {
    userRoles: getUserAccessData(state),
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    createMineRecord
  }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(CreateGuard(CreateMine));
