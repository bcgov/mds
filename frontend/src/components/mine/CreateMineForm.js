import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { createMineRecord } from '../../actionCreators/mineActionCreator';
import { Form, Input, Button } from 'antd';

const FormItem = Form.Item;

// follow https://ant.design/components/form/
export class CreateMineForm extends Component {

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.createMineRecord();
  }

  render() {
    return (
      <Form onSubmit={this.handleSubmit}>
        <FormItem>
          <Input></Input>
          <Button type="primary" htmlType="submit">
            Create Mine
          </Button>
        </FormItem>
      </Form>
    );
  }
}

const WrappedCreateMineForm = Form.create()(CreateMineForm);

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    createMineRecord
  }, dispatch);
}

export default connect(null, mapDispatchToProps)(WrappedCreateMineForm);
