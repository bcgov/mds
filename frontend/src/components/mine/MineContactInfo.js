
/**
 * @class MineContactInfo.js contains all information under the 'Contact Information' tab on the MnieDashboard (including all Mine Manager information);
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  mine: PropTypes.object.isRequired,
};

const defaultProps = {
  mine: {},
};


class MineContactInfo extends Component {
  render() {
    return (
      <div>
        <h1>contact information</h1>
      </div>
    );
  }
}

MineContactInfo.propTypes = propTypes;
MineContactInfo.defaultProps = defaultProps;

export default MineContactInfo;