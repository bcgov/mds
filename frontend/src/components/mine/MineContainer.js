/**
 * @class MineContainer.js is the connected container which gets a single mine record and passes it down to  MineDashboard.js..
 */
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { getMineRecord, updateMineRecord } from '@/actionCreators/mineActionCreator';
import { getMines, getMineIds } from '@/selectors/mineSelectors';
import Loading from '@/components/reusables/Loading';
import MineDashboard from '@/components/mine/MineDashboard';


const propTypes = {
  getMineRecord: PropTypes.func,
  updateMineRecord: PropTypes.func,
  mines: PropTypes.object,
  mineIds: PropTypes.array,
};

const defaultProps = {
  mines: {},
  mineIds: [],
};

class MineContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      mineId: ''
    }
  }

  componentDidMount() {
    const { id } = this.props.match.params;
    this.setState({ mineId: id })
    this.props.getMineRecord(id);
  }

  render() {
    if (this.state.mineId) {
      return (
        this.props.mineIds.map((id) => {
          return (
            <div key={id}>
              <MineDashboard mine={this.props.mines[id]} {...this.props} {...this.state}/>
            </div>
          );
        })
      )
    } else {
      return (<Loading />)
    }
  }
}


const mapStateToProps = (state) => {
  return {
    mines: getMines(state),
    mineIds: getMineIds(state),
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    getMineRecord,
    updateMineRecord
  }, dispatch);
};

MineContainer.propTypes = propTypes;
MineContainer.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(MineContainer);