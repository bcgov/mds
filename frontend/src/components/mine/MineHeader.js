import React, { Component } from "react";
import PropTypes from "prop-types";
import MineMap from "@/components/maps/MineMap";
import { Menu, Divider, Button, Dropdown, Tag } from "antd";
import { ELLIPSE, BRAND_PENCIL, RED_ELLIPSE, BRAND_DOCUMENT, EDIT } from "@/constants/assets";
import * as String from "@/constants/strings";
import * as ModalContent from "@/constants/modalContent";
import { modalConfig } from "@/components/modalContent/config";
import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
import CustomPropTypes from "@/customPropTypes";
import * as Permission from "@/constants/permissions";

/**
 * @class MineHeader.js contains header section of MineDashboard before the tabs. Including map, mineName, mineNumber.
 */
const propTypes = {
  closeModal: PropTypes.func.isRequired,
  openModal: PropTypes.func.isRequired,
  updateMineRecord: PropTypes.func.isRequired,
  removeMineType: PropTypes.func.isRequired,
  createTailingsStorageFacility: PropTypes.func.isRequired,
  fetchMineRecordById: PropTypes.func.isRequired,
  mine: CustomPropTypes.mine.isRequired,
  mineRegionHash: PropTypes.objectOf(PropTypes.string).isRequired,
  mineTenureHash: PropTypes.objectOf(PropTypes.string).isRequired,
  mineDisturbanceOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
  mineCommodityOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
  transformedMineTypes: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.strings)).isRequired,
};

class MineHeader extends Component {
  handleUpdateMineRecord = (value) => {
    const mineStatus = value.mine_status.join(",");
    this.props
      .updateMineRecord(
        this.props.mine.guid,
        { ...value, mine_status: mineStatus, mineType: this.props.mine.mine_type },
        value.name
      )
      .then(() => {
        this.props.closeModal();
        this.props.fetchMineRecordById(this.props.mine.guid);
      });
  };

  handleDeleteMineType = (event, mineTypeCode) => {
    event.preventDefault();
    this.props.mine.mine_type.map((type) => {
      if (type.mine_tenure_type_code === mineTypeCode) {
        const tenure = this.props.mineTenureHash[mineTypeCode];
        this.props.removeMineType(type.mine_type_guid, tenure).then(() => {
          this.props.fetchMineRecordById(this.props.mine.guid);
        });
      }
    });
  };

  handleAddTailings = (value) => {
    this.props
      .createTailingsStorageFacility({
        ...value,
        mine_guid: this.props.mine.guid,
      })
      .then(() => {
        this.props.closeModal();
        this.props.fetchMineRecordById(this.props.mine.guid);
      });
  };

  openTailingsModal(event, onSubmit, title) {
    event.preventDefault();
    this.props.openModal({
      props: { onSubmit, title },
      content: modalConfig.ADD_TAILINGS,
    });
  }

  openModal(event, onSubmit, handleDelete, title, mine) {
    event.preventDefault();
    const initialValues = {
      name: mine.mine_name ? mine.mine_name : null,
      latitude: mine.mine_location ? mine.mine_location.latitude : null,
      longitude: mine.mine_location ? mine.mine_location.longitude : null,
      mine_status: mine.mine_status[0] ? mine.mine_status[0].status_values : null,
      major_mine_ind: mine.major_mine_ind ? mine.major_mine_ind : false,
      mine_region: mine.region_code ? mine.region_code : null,
    };

    this.props.openModal({
      props: {
        onSubmit,
        handleDelete,
        title,
        initialValues,
      },
      content: modalConfig.MINE_RECORD,
      clearOnSubmit: false,
    });
  }

  render() {
    const menu = (
      <Menu>
        <Menu.Item key="0">
          <button
            type="button"
            className="full"
            onClick={(event) =>
              this.openModal(
                event,
                this.handleUpdateMineRecord,
                this.handleDeleteMineType,
                ModalContent.UPDATE_MINE_RECORD,
                this.props.mine
              )
            }
          >
            <img alt="pencil" className="padding-small" src={BRAND_PENCIL} />
            {ModalContent.UPDATE_MINE_RECORD}
          </button>
        </Menu.Item>
        <Menu.Item key="1">
          <button
            type="button"
            className="full"
            onClick={(event) =>
              this.openTailingsModal(event, this.handleAddTailings, ModalContent.ADD_TAILINGS)
            }
          >
            <img alt="document" className="padding-small" src={BRAND_DOCUMENT} />
            {ModalContent.ADD_TAILINGS}
          </button>
        </Menu.Item>
      </Menu>
    );
    return (
      <div className="dashboard__header--card">
        <div className="dashboard__header--card__content">
          <div className="inline-flex between center-mobile">
            <h1>{this.props.mine.mine_name} </h1>
            <AuthorizationWrapper
              permission={Permission.CREATE}
              isMajorMine={this.props.mine.major_mine_ind}
            >
              <Dropdown className="full-height full-mobile" overlay={menu} placement="bottomLeft">
                <Button type="primary">
                  <div className="padding-small">
                    <img className="padding-small--right" src={EDIT} alt="Add/Edit" />
                    Add/Edit
                  </div>
                </Button>
              </Dropdown>
            </AuthorizationWrapper>
          </div>
          <Divider className="custom-large-divider" />
          <div className="inline-flex between block-mobile">
            <div className="inline-flex padding-small">
              <p className="field-title">Mine No. </p>
              <p> {this.props.mine.mine_no} </p>
            </div>
            <div className="inline-flex padding-small">
              <p className="field-title">Mine Class </p>
              <p>{this.props.mine.major_mine_ind ? String.MAJOR_MINE : String.REGIONAL_MINE}</p>
            </div>
            <div className="inline-flex padding-small">
              <p className="field-title">TSF</p>
              <p>
                {this.props.mine.mine_tailings_storage_facility.length > 0
                  ? this.props.mine.mine_tailings_storage_facility.length
                  : String.EMPTY_FIELD}
              </p>
            </div>
          </div>
          {this.props.mine.mine_status[0] && (
            <div className="inline-flex padding-small">
              <p className="field-title">Operating Status </p>
              <img
                alt="status"
                className="dashboard__header--card__content--status__img"
                src={
                  this.props.mine.mine_status[0].status_values[0] === "OP" ? ELLIPSE : RED_ELLIPSE
                }
              />
              {this.props.mine.mine_status[0] ? (
                this.props.mine.mine_status[0].status_labels.map((label) => (
                  <p className="mine__status" key={label}>
                    {label}
                  </p>
                ))
              ) : (
                <p>{String.EMPTY_FIELD}</p>
              )}
            </div>
          )}
          {!this.props.mine.mine_status[0] && (
            <div className="inline-flex padding-small">
              <p className="field-title">Operating Status</p>
              <p>{String.EMPTY_FIELD}</p>
            </div>
          )}
          <div className="inline-flex padding-small">
            <p className="field-title">Tenure</p>
            <p>
              {this.props.transformedMineTypes.mine_tenure_type_code.length > 0 ? (
                this.props.transformedMineTypes.mine_tenure_type_code.map((tenure) => (
                  <span className="mine_tenure" key={tenure}>
                    {this.props.mineTenureHash[tenure]}
                  </span>
                ))
              ) : (
                <p>{String.EMPTY_FIELD}</p>
              )}
            </p>
          </div>
          <div className="inline-flex padding-small wrap">
            <p className="field-title">Commodity</p>
            {this.props.transformedMineTypes.mine_commodity_code.length > 0 ? (
              this.props.transformedMineTypes.mine_commodity_code.map((code) => (
                <Tag key={code}>{this.props.mineCommodityOptionsHash[code]}</Tag>
              ))
            ) : (
              <p>{String.EMPTY_FIELD}</p>
            )}
          </div>
          <div className="inline-flex padding-small wrap">
            <p className="field-title">Disturbance</p>
            {this.props.transformedMineTypes.mine_disturbance_code.length > 0 ? (
              this.props.transformedMineTypes.mine_disturbance_code.map((code) => (
                <Tag key={code}>{this.props.mineDisturbanceOptionsHash[code]}</Tag>
              ))
            ) : (
              <p>{String.EMPTY_FIELD}</p>
            )}
          </div>
        </div>
        <div className="dashboard__header--card__map">
          <MineMap mine={this.props.mine} />
          <div className="dashboard__header--card__map--footer">
            <div className="inline-flex between">
              <p className="p-white">
                Lat:{" "}
                {this.props.mine.mine_location
                  ? this.props.mine.mine_location.latitude
                  : String.EMPTY_FIELD}
              </p>
              <p className="p-white">
                Long:{" "}
                {this.props.mine.mine_location
                  ? this.props.mine.mine_location.longitude
                  : String.EMPTY_FIELD}
              </p>
            </div>
            <p className="p-white">
              Region:{" "}
              {this.props.mine.region_code
                ? this.props.mineRegionHash[this.props.mine.region_code]
                : String.EMPTY_FIELD}
            </p>
          </div>
        </div>
      </div>
    );
  }
}

MineHeader.propTypes = propTypes;

export default MineHeader;
