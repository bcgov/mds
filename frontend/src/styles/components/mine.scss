.mine__dashboard {
  &__header {
    &__img {
      height: 40vh;
      width: 100vw;
    }
  }
}

.ant-drawer {
  z-index: 100;
}