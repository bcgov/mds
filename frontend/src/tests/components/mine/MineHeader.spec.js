import React from "react";
import { shallow } from "enzyme";
import { MineHeader } from "@/components/mine/MineHeader";
import * as MOCK from "@/tests/mocks/dataMocks";

const props = {};
const dispatchProps = {};

const setupDispatchProps = () => {
  dispatchProps.updateMineRecord = jest.fn();
  dispatchProps.fetchMineRecordById = jest.fn();
  dispatchProps.closeModal = jest.fn();
  dispatchProps.openModal = jest.fn();
};

const setupProps = () => {
  props.mine = MOCK.MINES.mines[MOCK.MINES.mineIds[0]];
  props.mineStatusOptions = MOCK.STATUS_OPTIONS.records;
  props.mineRegionOptions = MOCK.REGION_DROPDOWN_OPTIONS;
  props.mineRegionHash = MOCK.REGION_HASH;
  props.mineTenureTypes = MOCK.TENURE_TYPES_DROPDOWN_OPTIONS;
  props.mineTenureHash = MOCK.TENURE_HASH;
  [props.transformedMineTypes] = MOCK.MINE_TYPES;
};

beforeEach(() => {
  setupDispatchProps();
  setupProps();
});

describe("MineHeader", () => {
  it("renders dispatchProperly", () => {
    const component = shallow(<MineHeader {...dispatchProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
});
