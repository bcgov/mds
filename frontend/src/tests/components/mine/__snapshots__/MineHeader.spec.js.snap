// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MineHeader renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MineHeader
    mine={
      Object {
        "guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
        "mgr_appointment": Array [],
        "mine_detail": Array [
          Object {
            "mine_name": "mine1",
            "mine_no": "BLAH9091",
          },
          Object {
            "mine_name": "mine3",
            "mine_no": "BLAH9091",
          },
        ],
        "mine_location": Array [
          Object {
            "latitude": null,
            "longitude": null,
          },
        ],
        "mineral_tenure_xref": Array [
          Object {
            "tenure_number_id": "1234567",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MineMap
          mine={
            Object {
              "guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
              "mgr_appointment": Array [],
              "mine_detail": Array [
                Object {
                  "mine_name": "mine1",
                  "mine_no": "BLAH9091",
                },
                Object {
                  "mine_name": "mine3",
                  "mine_no": "BLAH9091",
                },
              ],
              "mine_location": Array [
                Object {
                  "latitude": null,
                  "longitude": null,
                },
              ],
              "mineral_tenure_xref": Array [
                Object {
                  "tenure_number_id": "1234567",
                },
              ],
            }
          }
        />,
        <div
          className="dashboard__header__content"
        >
          <h1>
            mine1
          </h1>
          <h5>
            Mine ID: 
            BLAH9091
             
          </h5>
          <div
            className="dashboard__header__content--inline"
          >
            <div
              className="inline-flex"
            >
              <img
                className="inline-flex--img"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <div>
                <p>
                  Lat:
                </p>
              </div>
              <div>
                <p>
                  Long:
                </p>
              </div>
            </div>
            <div
              className="inline-flex"
            >
              <img
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              <div>
                <h5>
                  Status: 
                </h5>
              </div>
              <div>
                <h3>
                  Active
                </h3>
              </div>
            </div>
          </div>
        </div>,
      ],
      "className": "dashboard__header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "mine": Object {
            "guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
            "mgr_appointment": Array [],
            "mine_detail": Array [
              Object {
                "mine_name": "mine1",
                "mine_no": "BLAH9091",
              },
              Object {
                "mine_name": "mine3",
                "mine_no": "BLAH9091",
              },
            ],
            "mine_location": Array [
              Object {
                "latitude": null,
                "longitude": null,
              },
            ],
            "mineral_tenure_xref": Array [
              Object {
                "tenure_number_id": "1234567",
              },
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              mine1
            </h1>,
            <h5>
              Mine ID: 
              BLAH9091
               
            </h5>,
            <div
              className="dashboard__header__content--inline"
            >
              <div
                className="inline-flex"
              >
                <img
                  className="inline-flex--img"
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
                <div>
                  <p>
                    Lat:
                  </p>
                </div>
                <div>
                  <p>
                    Long:
                  </p>
                </div>
              </div>
              <div
                className="inline-flex"
              >
                <img
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
                <div>
                  <h5>
                    Status: 
                  </h5>
                </div>
                <div>
                  <h3>
                    Active
                  </h3>
                </div>
              </div>
            </div>,
          ],
          "className": "dashboard__header__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mine1",
            },
            "ref": null,
            "rendered": "mine1",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Mine ID: ",
                "BLAH9091",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Mine ID: ",
              "BLAH9091",
              " ",
            ],
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="inline-flex"
                >
                  <img
                    className="inline-flex--img"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                  <div>
                    <p>
                      Lat:
                    </p>
                  </div>
                  <div>
                    <p>
                      Long:
                    </p>
                  </div>
                </div>,
                <div
                  className="inline-flex"
                >
                  <img
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                  <div>
                    <h5>
                      Status: 
                    </h5>
                  </div>
                  <div>
                    <h3>
                      Active
                    </h3>
                  </div>
                </div>,
              ],
              "className": "dashboard__header__content--inline",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      className="inline-flex--img"
                      src={
                        Object {
                          "process": [Function],
                        }
                      }
                    />,
                    <div>
                      <p>
                        Lat:
                      </p>
                    </div>,
                    <div>
                      <p>
                        Long:
                      </p>
                    </div>,
                  ],
                  "className": "inline-flex",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "inline-flex--img",
                      "src": Object {
                        "process": [Function],
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p>
                        Lat:
                      </p>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Lat:",
                          null,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Lat:",
                        null,
                      ],
                      "type": "p",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p>
                        Long:
                      </p>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Long:",
                          null,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Long:",
                        null,
                      ],
                      "type": "p",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      src={
                        Object {
                          "process": [Function],
                        }
                      }
                    />,
                    <div>
                      <h5>
                        Status: 
                      </h5>
                    </div>,
                    <div>
                      <h3>
                        Active
                      </h3>
                    </div>,
                  ],
                  "className": "inline-flex",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "src": Object {
                        "process": [Function],
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h5>
                        Status: 
                      </h5>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Status: ",
                      },
                      "ref": null,
                      "rendered": "Status: ",
                      "type": "h5",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h3>
                        Active
                      </h3>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Active",
                      },
                      "ref": null,
                      "rendered": "Active",
                      "type": "h3",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MineMap
            mine={
              Object {
                "guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
                "mgr_appointment": Array [],
                "mine_detail": Array [
                  Object {
                    "mine_name": "mine1",
                    "mine_no": "BLAH9091",
                  },
                  Object {
                    "mine_name": "mine3",
                    "mine_no": "BLAH9091",
                  },
                ],
                "mine_location": Array [
                  Object {
                    "latitude": null,
                    "longitude": null,
                  },
                ],
                "mineral_tenure_xref": Array [
                  Object {
                    "tenure_number_id": "1234567",
                  },
                ],
              }
            }
          />,
          <div
            className="dashboard__header__content"
          >
            <h1>
              mine1
            </h1>
            <h5>
              Mine ID: 
              BLAH9091
               
            </h5>
            <div
              className="dashboard__header__content--inline"
            >
              <div
                className="inline-flex"
              >
                <img
                  className="inline-flex--img"
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
                <div>
                  <p>
                    Lat:
                  </p>
                </div>
                <div>
                  <p>
                    Long:
                  </p>
                </div>
              </div>
              <div
                className="inline-flex"
              >
                <img
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
                <div>
                  <h5>
                    Status: 
                  </h5>
                </div>
                <div>
                  <h3>
                    Active
                  </h3>
                </div>
              </div>
            </div>
          </div>,
        ],
        "className": "dashboard__header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "mine": Object {
              "guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
              "mgr_appointment": Array [],
              "mine_detail": Array [
                Object {
                  "mine_name": "mine1",
                  "mine_no": "BLAH9091",
                },
                Object {
                  "mine_name": "mine3",
                  "mine_no": "BLAH9091",
                },
              ],
              "mine_location": Array [
                Object {
                  "latitude": null,
                  "longitude": null,
                },
              ],
              "mineral_tenure_xref": Array [
                Object {
                  "tenure_number_id": "1234567",
                },
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                mine1
              </h1>,
              <h5>
                Mine ID: 
                BLAH9091
                 
              </h5>,
              <div
                className="dashboard__header__content--inline"
              >
                <div
                  className="inline-flex"
                >
                  <img
                    className="inline-flex--img"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                  <div>
                    <p>
                      Lat:
                    </p>
                  </div>
                  <div>
                    <p>
                      Long:
                    </p>
                  </div>
                </div>
                <div
                  className="inline-flex"
                >
                  <img
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                  <div>
                    <h5>
                      Status: 
                    </h5>
                  </div>
                  <div>
                    <h3>
                      Active
                    </h3>
                  </div>
                </div>
              </div>,
            ],
            "className": "dashboard__header__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mine1",
              },
              "ref": null,
              "rendered": "mine1",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Mine ID: ",
                  "BLAH9091",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Mine ID: ",
                "BLAH9091",
                " ",
              ],
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="inline-flex"
                  >
                    <img
                      className="inline-flex--img"
                      src={
                        Object {
                          "process": [Function],
                        }
                      }
                    />
                    <div>
                      <p>
                        Lat:
                      </p>
                    </div>
                    <div>
                      <p>
                        Long:
                      </p>
                    </div>
                  </div>,
                  <div
                    className="inline-flex"
                  >
                    <img
                      src={
                        Object {
                          "process": [Function],
                        }
                      }
                    />
                    <div>
                      <h5>
                        Status: 
                      </h5>
                    </div>
                    <div>
                      <h3>
                        Active
                      </h3>
                    </div>
                  </div>,
                ],
                "className": "dashboard__header__content--inline",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        className="inline-flex--img"
                        src={
                          Object {
                            "process": [Function],
                          }
                        }
                      />,
                      <div>
                        <p>
                          Lat:
                        </p>
                      </div>,
                      <div>
                        <p>
                          Long:
                        </p>
                      </div>,
                    ],
                    "className": "inline-flex",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "inline-flex--img",
                        "src": Object {
                          "process": [Function],
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <p>
                          Lat:
                        </p>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "Lat:",
                            null,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          "Lat:",
                          null,
                        ],
                        "type": "p",
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <p>
                          Long:
                        </p>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "Long:",
                            null,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          "Long:",
                          null,
                        ],
                        "type": "p",
                      },
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        src={
                          Object {
                            "process": [Function],
                          }
                        }
                      />,
                      <div>
                        <h5>
                          Status: 
                        </h5>
                      </div>,
                      <div>
                        <h3>
                          Active
                        </h3>
                      </div>,
                    ],
                    "className": "inline-flex",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "src": Object {
                          "process": [Function],
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <h5>
                          Status: 
                        </h5>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Status: ",
                        },
                        "ref": null,
                        "rendered": "Status: ",
                        "type": "h5",
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <h3>
                          Active
                        </h3>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Active",
                        },
                        "ref": null,
                        "rendered": "Active",
                        "type": "h3",
                      },
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
